code/obj/item/device/nuclear_op_items.dm

```diff
@@ -0,0 +1,824 @@
+		sleep(rand(60, 110))
+		if(!isnull(src.target_overlay))
+			target_turf.overlays -= src.target_overlay
+		explosion_new(user, target_turf, 75)
+		return TRUE
+
+
+	syndicate
+		firingfrom = "Cairngorm"
+
+#undef DESIGNATOR_MAX_RANGE
+////////// Ammoboxes & ammobags //////////
+//Not all used for nukeops, but theme stays the same throughout so they're all here
+
+/obj/item/ammo/ammobox
+	sname = "Generic Ammobox"
+	name = "Generic Ammobox"
+	desc = "A generic ammobox for getting some ammunition."
+	icon_state = "lmg_ammo-0-old"
+	ammo_type = null
+	caliber = null
+	var/list/valid_calibers = list() //supports lists and single, set to "All" for any gun
+
+	attackby(obj/item/gun/kinetic/W, mob/user)
+		if(istype(W, /obj/item/gun/kinetic))
+			if((islist(valid_calibers) && initial(W.caliber) in valid_calibers || (!islist(valid_calibers) && valid_calibers == initial(W.caliber)) || valid_calibers == "All")
+				new W.default_magazine(get_turf(src))
+				var/obj/O = W.default_magazine
+				boutput(user, "<span class='alert'>You get a [O.name] out of [src].</span>")
+				qdel(src)
+			if(valid_calibers == "All")
+				new W.default_magazine(get_turf(src))
+				var/obj/O = W.default_magazine
+				boutput(user, "<span class='alert'>You get a [O.name] out of [src].</span>")
+				qdel(src)
+		else
+			..()
+
+/obj/item/ammo/ammobox/nukeop
+	name = "Syndicate Ammo Bag"
+	desc = "A bag that can fabricate magazines for standard syndicate weapons. Technology!"
+	icon_state = "ammobag"
+	item_state = "ammobag"
+	var/charge = 10
+	var/spec_ammo = FALSE
+	var/deployed = FALSE
+
+	New()
+		..()
+		if(!deployed)
+			src.desc = "A folded up bag that, once deployed, can fabricate magazines for standard syndicate weapons. It has [src.charge] charge left."
+		else
+			src.desc = "A bag that can fabricate magazines for standard syndicate weapons. Technology! It has [src.charge] charge left."
+
+	attack_self(mob/user)
+		if(!deployed)
+			user.visible_message("[user] begins unfolding a [src].", "You begin unfolding \the [src].")
+			SETUP_GENERIC_ACTIONBAR(user, src, 5 SECONDS, /obj/item/ammo/ammobox/nukeop/proc/deploy_ammobag, user, src.icon, src.icon_state,"[user] finishes deploying a [src].", null)
+
+	MouseDrop(atom/over_object, src_location, over_location, over_control, params)
+		if(!(over_object == usr))
+			return
+
+		if(deployed)
+			usr.visible_message("[usr] begins folding up [src].", "You begin folding up \the [src].")
+			SETUP_GENERIC_ACTIONBAR(usr, src, 5 SECONDS, /obj/item/ammo/ammobox/nukeop/proc/fold_ammobag, usr, src.icon, src.icon_state,"[usr] finishes folding up [src].", null)
+
+
+	proc/deploy_ammobag(var/mob/user)
+		force_drop(user)
+		anchored = TRUE
+		deployed = TRUE
+		icon_state = "[initial(icon_state)]-d[charge <= 0 ? "-empty" : ""]"
+
+	proc/fold_ammobag(var/mob/user)
+		anchored = FALSE
+		deployed = FALSE
+		icon_state = "[initial(icon_state)]"
+		sleep(1 DECI SECOND)
+		Attackhand(user)
```

Comment:
what is this

Reply:
A way to automatically pick it up after folding it up